src/git-auth-helper.ts

```diff
@@ -19,8 +19,9 @@ export interface IGitAuthHelper {
   configureAuth(): Promise<void>
   configureGlobalAuth(): Promise<void>
   configureSubmoduleAuth(): Promise<void>
+  configureTempGlobalConfig(repositoryPath?: string): Promise<void>
```

Comment:
what about making `configureTempGlobalConfig` return the path to the temp config file?

Reply:
so we don't need to do `const newGitConfigPath = path.join(this.temporaryHomePath, '.gitconfig')` twice?